packages/wallet/test/integration/buildTx.test.ts

```diff
@@ -0,0 +1,143 @@
+/* eslint-disable func-style */
+/* eslint-disable jsdoc/require-jsdoc */
+import * as mocks from '../mocks';
+import { Cardano } from '@cardano-sdk/core';
+import {
+  MaybeValidTx,
+  MaybeValidTxOut,
+  ObservableWallet,
+  OutputValidation,
+  OutputValidationError,
+  OutputValidator,
+  TxOutputFailure,
+  ValidTx,
+  ValidTxOut,
+  buildTx
+} from '../../src';
+import { createWallet } from './util';
+
+function assertTxIsValid(tx: MaybeValidTx): asserts tx is ValidTx {
+  expect(tx.isValid).toBe(true);
+}
+function assertTxOutIsValid(txOut: MaybeValidTxOut): asserts txOut is ValidTxOut {
+  expect(txOut.isValid).toBe(true);
+}
+
+describe('buildTx', () => {
+  let wallet: ObservableWallet;
+
+  beforeEach(async () => {
+    ({ wallet } = await createWallet());
+  });
+
+  afterEach(() => wallet.shutdown());
+
+  it('can be used to build, sign and submit a tx', async () => {
+    const tx = await buildTx(wallet).addOutput(mocks.utxo[0][1]).build();
+    if (tx.isValid) {
+      const signedTx = await tx.sign();
+      await signedTx.submit();
+    } else {
+      expect(tx.errors.length).toBeGreaterThan(0);
+      throw new Error('Invalid tx');
+    }
+  });
+
+  it('can add outputs one by one', async () => {
+    const tx = await buildTx(wallet).addOutput(mocks.utxo[0][1]).addOutput(mocks.utxo[1][1]).build();
+    assertTxIsValid(tx);
+    // 2 outputs + change output
+    expect(tx.body.outputs.length).toBe(3);
+  });
+
+  it('can remove outputs and build', async () => {
+    const txBuilder = buildTx(wallet);
+    const txTwoOutputs = txBuilder.addOutput(mocks.utxo[0][1]).addOutput(mocks.utxo[1][1]);
+    const tx = await txTwoOutputs.removeOutput(mocks.utxo[0][1]).build();
+    assertTxIsValid(tx);
+    expect(tx.body.outputs.length).toBe(2);
+  });
+
+  it('builds transactions that are not modified by subsequent builder changes', async () => {
```

Comment:
I'm not following based on the test content. Is there maybe a missing assertion between line 63 and 64?

Reply:
Huge improvement, well done!